### ISSUE TITLE
[Feature][plugins] Add `connection_id` to `transformation_rule`

### ISSUE THREAD

**Issue by klesh** [apache/incubator-devlake#4681]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

Users are able to edit jira transformation rules

### Description

Currently, transformation rules are independent, they are top-level entities of a plugin. This poses a serious problem for JIRA:
![img_v2_376cd9fb-84f2-4043-a160-4e86d736433g](https://user-images.githubusercontent.com/61080/225506801-a37a86e3-0b4d-4c67-a624-bbafbfa67a58.jpg)
From the above screenshot, we can see the configuration process needs to load data from the jira API, thus `connection_id` is required.

And we can safely assume that it is true for other plugins.
In other words, **a `transformation rule` must belong to a specific `connection`**

We have to add `connection_id` to `transformation rule` for all our plugins

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)
